semver.py

```diff
@@ -47,10 +28,44 @@ def cmp(a, b):
         return (a > b) - (a < b)
 
 
+def deprecated(replace=None):
+    """
+    Decorates a function to output a deprecation warning.
+
+    This function will be removed once major version 3 of semver is
+    released.
+    """
+    # we can't use the 'nonlocal' keyword in Python2, so we need
+    # to circumvent that with a dictionary:
+    r = {"r": replace}
```

Comment:
`nonlocal` is needed only when you reassign a variable (`replace = ...`), not when you just access it. Thus lines 40 and 46 can leave and replace `r=replace` in line 52 with `r=replace or func.__name__`

Reply:
True, I was mislead by this nonlocal keyword. Replaced it with your suggestion. :+1: